numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -1349,32 +1369,52 @@ string_replace_strided_loop(
             }
         }
 
-        // conservatively overallocate
-        // TODO check overflow
-        size_t max_size;
-        if (i2s.size == 0) {
-            // interleaving
-            max_size = i1s.size + (i1s.size + 1)*(i3s.size);
-        }
-        else {
-            // replace i2 with i3
-            max_size = i1s.size * (i3s.size/i2s.size + 1);
-        }
-        char *new_buf = (char *)PyMem_RawCalloc(max_size, 1);
-        Buffer<ENCODING::UTF8> buf1((char *)i1s.buf, i1s.size);
-        Buffer<ENCODING::UTF8> buf2((char *)i2s.buf, i2s.size);
-        Buffer<ENCODING::UTF8> buf3((char *)i3s.buf, i3s.size);
-        Buffer<ENCODING::UTF8> outbuf(new_buf, max_size);
+        {
```

Comment:
Why the new indentation? It already is in the loop.

(And it makes reviewing harder...)

Reply:
It's because of the new use of `goto next_step`, I need to defined a new lexical scope or define a bunch of variables at the top of the for loop that are only used at the bottom of it.

Reply:
I'd probably have gone for top of the for-loop myself, but no big deal...

Reply:
While I don't hate the `while (N--)` loop in general, I do think a `goto` for loop control flow isn't nice and I much prefer a long `for` instead.
But this file has this pattern in a few places right now so it doesn't matter since the other places use this pattern also.